src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,335 @@
+ * The indexer will use IR sensors so we can see the presence of a ball The
+ * distance that the balls are pushed between the belts needs to be coded so
+ * that they can move fluidly When a ball is at the top of the Pierre, the ball
+ * at the entrance of Pierre needs to be moved forward at the same pace of the
+ * balls being intaked
+ * 
+ */
+
+public class Indexer extends SubsystemBase {
+    final SendableSpeedController singulator;
+    private final SendableSpeedController pierreMotor;
+    final BooleanSupplier frontIntakeIR;
+    final BooleanSupplier bottomPierreIR;
+    final BooleanSupplier topPierreIR;
+    final BooleanSupplier backIntakeIR;
+    public double ballCounting;
+
+    private static final double singulatorMotorSpeed = 0.95;
+    private static final double pierreIndexSpeed = 0.85;
+
+
+
+    public Indexer(final IndexMap map) {
+        super();
+        singulator = map.singulator();
+        frontIntakeIR = map.frontIntakeIR();
+        bottomPierreIR = map.bottomPierreIR();
+        topPierreIR = map.topPierreIR();
+        backIntakeIR = map.backIntakeIR();
+
+        // bottomPierreIR = map.bottomPierreIR();
+        // topPierreIR = map.topPierreIR();
+        // backIntakeIR = map.backIntakeIR();
+        pierreMotor = map.pierreMotor();
+        
+    }
+
+    public SequentialCommandGroup intakeToPierre() {
+        return new SequentialCommandGroup(pierrePossesion(), runToClear());
+    }
+    
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered(){
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(intakeToPierre(), pierrePossesion(),
+                singulatorPossesion(),
+                loadBallToTop(), runToClear());
+
+
+         topIRStop.setDeadline(stopWhenBallsAtTop());
+
+         return topIRStop;
+    }
+    //This will stop all commands when the top IR sensor on Pierre is triggered
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        });
+    }
+
+    public CommandBase runPierre() {
+        return new StartEndCommand(() -> {
+            pierreMotor.set(.75);
+        }, () -> {
+            pierreMotor.stopMotor();
+        }, this);
+    }
+
+
+
+    public CommandBase quicklyPush() {
+        return indexMotor(singulatorMotorSpeed);
+    }
+
+    public CommandBase reversePush() {
+        return indexMotor(-singulatorMotorSpeed);
+    }
+
+    public CommandBase quicklyOutput() {
+        return indexMotor(pierreIndexSpeed);
+    }
+
+    public CommandBase reverseOutput() {
+        return indexMotor(-pierreIndexSpeed);
+    }
+
+    /*
+     * Command to run the singlator motor until the IR sensor 1 (analog 0) Command
```

Comment:
It looks like this first sentence ends prematurely. Use the new sensor names instead of `IR senor 1` (there's also a spelling mistake).